### PR TITLE
fix(treeView): fix sql query to get correct status

### LINKED ISSUE #605: The build counts in some trees in Tree Listing Page don't match with Tree Details counts
Go to [Tree Listing Page](https://staging.dashboard.kernelci.org:9000/tree)
Go to this [Tree Details](https://staging.dashboard.kernelci.org:9000/tree/38dc270ca093908e056ebb9d1f60ce7b7a654b5e?tableFilter=%7B%22bootsTable%22%3A%22all%22%2C%22buildsTable%22%3A%22all%22%2C%22testsTable%22%3A%22all%22%7D&origin=maestro&currentPageTab=treeDetails.builds&diffFilter=%7B%7D&treeInfo=%7B%22gitUrl%22%3A%22https%3A%2F%2Fandroid.googlesource.com%2Fkernel%2Fcommon%22%2C%22gitBranch%22%3A%22android12-5.10-lts%22%2C%22treeName%22%3A%22android%22%2C%22commitName%22%3A%22ASB-2024-11-05_12-5.10-668-g38dc270ca093%22%2C%22headCommitHash%22%3A%2238dc270ca093908e056ebb9d1f60ce7b7a654b5e%22%7D&intervalInDays=7)

Compare the counts of builds in both pages.

### PR DESCRIPTION
**Pull request by murilx**
Change the SQL query in the TreeView. Before It was considering only the latest checkout when counting the status por builds, boots and tests that was causing some status to be incorrect. This new query groups the rows by the commit hash instead of the checkout ID so multiple checkouts that correspond to the same tree now are used when counting the status, returning the correct value

Closes #605

## Visual Reference
Before
versao_antiga

After
versao_atual

### CODE REVIEW

File: backend/kernelCI_app/views/treeView.py [hunk 1]
```diff
@@ -27,35 +27,19 @@ def get(self, request):
             "interval_param": getQueryTimeInterval(**interval_days_data).timestamp()
         }
 
+        # '1 as id' is necessary in this case because django raw queries must include the primary key.
+        # In this case we don't need the primary key and adding it would alter the GROUP BY clause,
+        # potentially causing the tree listing page show the same tree multiple times
```
Comment (WilsonNet) [inline backend/kernelCI_app/views/treeView.py:32]: sometimes we gotta do what we gotta do

File: backend/kernelCI_app/views/treeView.py [hunk 2]
```diff
@@ -27,35 +27,19 @@ def get(self, request):
...
-                            start_time DESC
-                    ) AS selection_sorted
-                GROUP BY
-                    git_repository_url, git_repository_branch
-                ORDER BY
-                    start_time DESC
-            )
-
             SELECT
-                checkouts.*,
```
Comment (WilsonNet) [inline backend/kernelCI_app/views/treeView.py:58]: thank god we removed this star query

**Review by WilsonNet — APPROVED** [review]
nice job, seems like the new query is working

File: backend/kernelCI_app/views/treeView.py [hunk 3]
```diff
@@ -89,25 +73,58 @@ def get(self, request):
...
+                        (
+                            SELECT
+                                id,
+                                tree_name,
+                                git_repository_branch,
+                                git_repository_url,
+                                git_commit_hash,
+                                git_commit_name,
+                                patchset_hash,
+                                start_time,
```
Comment (MarceloRobert) [inline backend/kernelCI_app/views/treeView.py:102]: do you need to select all these things for this subquery? `ordered_checkouts_by_tree` doesn't seem to be used anywhere else

**Review by MarceloRobert — COMMENTED** [review]
Could you explain what you're changing in the query in the PR or commit message so that it becomes clearer for other devs?

File: backend/kernelCI_app/views/treeView.py (hunk 3 above)
Comment (murilx) [reply backend/kernelCI_app/views/treeView.py]: Some of then were indeed not necessary (as far as I tested) so I removed them

**Review by MarceloRobert — APPROVED** [review]
Great work, LGTM

**Pull request merged**